没有libsimple依赖

.load libsimple 报错
macos环境下编译出sqlite3-shell后执行.load libsimple或.load libsimple.so 或.load libsimple.dylib报错
已经加入了LIB = -ldl -lpthread -lm

make libsimple.dylib                                                                                
rm -f libsimple.so
g++ -std=c++14 -stdlib=libc++ src/*.cc src/*.c -shared -rdynamic -o libsimple.dylib -I build/ -fPIC -ldl -lpthread -lm -DSQLITE_DEBUG -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_EXPLAIN_COMMENTS -DSQLITE_ENABLE_SELECTTRACE -DSQLITE_ENABLE_WHERETRACE
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:657:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:305:9: error: implicit instantiation of undefined template
      'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    _T1 first;
        ^
src/pinyin.h:17:7: note: in instantiation of template class 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> >, char>' requested here
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:423:45: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &, const char &>' requested
      here
            _CheckArgsDep<_Dummy>::template __enable_implicit<_T1 const&, _T2 const&>()
                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:426:5: note: while substituting prior template arguments
      into non-type template parameter [with _Dummy = true]
    pair(_T1 const& __t1, _T2 const& __t2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _Dummy = (no value), $1 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, char const (&)[3]>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:441:34: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<char const (&)[3], char>' requested here
            _CheckArgs::template __enable_implicit<_U1, _U2>()
                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:444:5: note: while substituting prior template arguments
      into non-type template parameter [with _U1 = char const (&)[3], _U2 = char]
    pair(_U1&& __u1, _U2&& __u2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _U1 = char const (&)[3], _U2 = char, $2 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:16:53: error: no matching constructor for initialization of 'const std::map<std::string, char>' (aka 'const map<basic_string<char, char_traits<char>,
      allocator<char> >, char>')
  const std::map<std::string, char> tone_to_plain = {
                                                    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:980:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:989:9: note: candidate constructor template not viable:
      requires 4 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:999:5: note: candidate constructor template not viable:
      requires 3 arguments, but 29 were provided
    map(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1053:5: note: candidate constructor not viable: requires 3
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:975:14: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    explicit map(const key_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1035:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(map&& __m, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1046:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1061:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1081:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(const map& __m, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:968:14: note: candidate constructor not viable: requires
      single argument '__comp', but 29 arguments were provided
    explicit map(const key_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1004:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(const map& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1029:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(map&& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1075:14: note: candidate constructor not viable: requires
      single argument '__a', but 29 arguments were provided
    explicit map(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:960:5: note: candidate constructor not viable: requires 0
      arguments, but 29 were provided
    map()
    ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:23:47: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_prefix = {
                                              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 63 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 63 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 63 were provided
    set()
    ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:45:46: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_valid = {
                                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 399 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 399 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 399 were provided
    set()
    ^
src/entry.cc:52:28: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
      std::string result = simple_tokenizer::SimpleTokenizer::tokenize_query(text, std::strlen(text));
                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
src/entry.cc:52:19: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
      std::string result = simple_tokenizer::SimpleTokenizer::tokenize_query(text, std::strlen(text));
                  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
8 errors generated.
In file included from src/pinyin.cc:1:
In file included from src/pinyin.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:479:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:14:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iterator:421:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__functional_base:18:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:305:9: error: implicit instantiation of undefined template
      'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    _T1 first;
        ^
src/pinyin.h:17:7: note: in instantiation of template class 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> >, char>' requested here
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/pinyin.cc:1:
In file included from src/pinyin.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:479:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:14:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iterator:421:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__functional_base:14:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:423:45: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &, const char &>' requested
      here
            _CheckArgsDep<_Dummy>::template __enable_implicit<_T1 const&, _T2 const&>()
                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:426:5: note: while substituting prior template arguments
      into non-type template parameter [with _Dummy = true]
    pair(_T1 const& __t1, _T2 const& __t2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _Dummy = (no value), $1 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/pinyin.cc:1:
In file included from src/pinyin.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:479:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:14:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iterator:421:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__functional_base:14:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, char const (&)[3]>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:441:34: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<char const (&)[3], char>' requested here
            _CheckArgs::template __enable_implicit<_U1, _U2>()
                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:444:5: note: while substituting prior template arguments
      into non-type template parameter [with _U1 = char const (&)[3], _U2 = char]
    pair(_U1&& __u1, _U2&& __u2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _U1 = char const (&)[3], _U2 = char, $2 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/pinyin.cc:1:
src/pinyin.h:16:53: error: no matching constructor for initialization of 'const std::map<std::string, char>' (aka 'const map<basic_string<char, char_traits<char>,
      allocator<char> >, char>')
  const std::map<std::string, char> tone_to_plain = {
                                                    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:980:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:989:9: note: candidate constructor template not viable:
      requires 4 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:999:5: note: candidate constructor template not viable:
      requires 3 arguments, but 29 were provided
    map(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1053:5: note: candidate constructor not viable: requires 3
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:975:14: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    explicit map(const key_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1035:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(map&& __m, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1046:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1061:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1081:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(const map& __m, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:968:14: note: candidate constructor not viable: requires
      single argument '__comp', but 29 arguments were provided
    explicit map(const key_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1004:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(const map& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1029:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(map&& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1075:14: note: candidate constructor not viable: requires
      single argument '__a', but 29 arguments were provided
    explicit map(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:960:5: note: candidate constructor not viable: requires 0
      arguments, but 29 were provided
    map()
    ^
In file included from src/pinyin.cc:1:
src/pinyin.h:23:47: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_prefix = {
                                              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 63 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 63 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 63 were provided
    set()
    ^
In file included from src/pinyin.cc:1:
src/pinyin.h:45:46: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_valid = {
                                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 399 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 399 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 399 were provided
    set()
    ^
6 errors generated.
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:657:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:305:9: error: implicit instantiation of undefined template
      'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    _T1 first;
        ^
src/pinyin.h:17:7: note: in instantiation of template class 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> >, char>' requested here
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:423:45: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &, const char &>' requested
      here
            _CheckArgsDep<_Dummy>::template __enable_implicit<_T1 const&, _T2 const&>()
                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:426:5: note: while substituting prior template arguments
      into non-type template parameter [with _Dummy = true]
    pair(_T1 const& __t1, _T2 const& __t2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _Dummy = (no value), $1 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, char const (&)[3]>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:441:34: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<char const (&)[3], char>' requested here
            _CheckArgs::template __enable_implicit<_U1, _U2>()
                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:444:5: note: while substituting prior template arguments
      into non-type template parameter [with _U1 = char const (&)[3], _U2 = char]
    pair(_U1&& __u1, _U2&& __u2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _U1 = char const (&)[3], _U2 = char, $2 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:16:53: error: no matching constructor for initialization of 'const std::map<std::string, char>' (aka 'const map<basic_string<char, char_traits<char>,
      allocator<char> >, char>')
  const std::map<std::string, char> tone_to_plain = {
                                                    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:980:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:989:9: note: candidate constructor template not viable:
      requires 4 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:999:5: note: candidate constructor template not viable:
      requires 3 arguments, but 29 were provided
    map(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1053:5: note: candidate constructor not viable: requires 3
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:975:14: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    explicit map(const key_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1035:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(map&& __m, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1046:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1061:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1081:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(const map& __m, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:968:14: note: candidate constructor not viable: requires
      single argument '__comp', but 29 arguments were provided
    explicit map(const key_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1004:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(const map& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1029:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(map&& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1075:14: note: candidate constructor not viable: requires
      single argument '__a', but 29 arguments were provided
    explicit map(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:960:5: note: candidate constructor not viable: requires 0
      arguments, but 29 were provided
    map()
    ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:23:47: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_prefix = {
                                              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 63 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 63 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 63 were provided
    set()
    ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:45:46: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_valid = {
                                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 399 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 399 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 399 were provided
    set()
    ^
6 errors generated.
make: *** [libsimple.dylib] Error 1

> > 你好，mac下面编译这个库的命令应该是make libsimple.dylib吧？
> 
> 对你试下；如果不行你告诉我我明天试试

你好，就是这种报错，看上去还像是c++版本问题？？怎么解决呢？
make libsimple.dylib                                                                                
rm -f libsimple.so
g++ -std=c++14 -stdlib=libc++ src/*.cc src/*.c -shared -rdynamic -o libsimple.dylib -I build/ -fPIC -ldl -lpthread -lm -DSQLITE_DEBUG -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_EXPLAIN_COMMENTS -DSQLITE_ENABLE_SELECTTRACE -DSQLITE_ENABLE_WHERETRACE
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:657:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:305:9: error: implicit instantiation of undefined template
      'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    _T1 first;
        ^
src/pinyin.h:17:7: note: in instantiation of template class 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> >, char>' requested here
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:423:45: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &, const char &>' requested
      here
            _CheckArgsDep<_Dummy>::template __enable_implicit<_T1 const&, _T2 const&>()
                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:426:5: note: while substituting prior template arguments
      into non-type template parameter [with _Dummy = true]
    pair(_T1 const& __t1, _T2 const& __t2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _Dummy = (no value), $1 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, char const (&)[3]>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:441:34: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<char const (&)[3], char>' requested here
            _CheckArgs::template __enable_implicit<_U1, _U2>()
                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:444:5: note: while substituting prior template arguments
      into non-type template parameter [with _U1 = char const (&)[3], _U2 = char]
    pair(_U1&& __u1, _U2&& __u2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _U1 = char const (&)[3], _U2 = char, $2 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:16:53: error: no matching constructor for initialization of 'const std::map<std::string, char>' (aka 'const map<basic_string<char, char_traits<char>,
      allocator<char> >, char>')
  const std::map<std::string, char> tone_to_plain = {
                                                    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:980:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:989:9: note: candidate constructor template not viable:
      requires 4 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:999:5: note: candidate constructor template not viable:
      requires 3 arguments, but 29 were provided
    map(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1053:5: note: candidate constructor not viable: requires 3
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:975:14: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    explicit map(const key_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1035:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(map&& __m, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1046:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1061:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1081:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(const map& __m, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:968:14: note: candidate constructor not viable: requires
      single argument '__comp', but 29 arguments were provided
    explicit map(const key_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1004:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(const map& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1029:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(map&& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1075:14: note: candidate constructor not viable: requires
      single argument '__a', but 29 arguments were provided
    explicit map(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:960:5: note: candidate constructor not viable: requires 0
      arguments, but 29 were provided
    map()
    ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:23:47: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_prefix = {
                                              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 63 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 63 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 63 were provided
    set()
    ^
In file included from src/entry.cc:2:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:45:46: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_valid = {
                                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 399 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 399 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 399 were provided
    set()
    ^
src/entry.cc:52:28: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
      std::string result = simple_tokenizer::SimpleTokenizer::tokenize_query(text, std::strlen(text));
                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
src/entry.cc:52:19: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
      std::string result = simple_tokenizer::SimpleTokenizer::tokenize_query(text, std::strlen(text));
                  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
8 errors generated.
In file included from src/pinyin.cc:1:
In file included from src/pinyin.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:479:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:14:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iterator:421:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__functional_base:18:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:305:9: error: implicit instantiation of undefined template
      'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    _T1 first;
        ^
src/pinyin.h:17:7: note: in instantiation of template class 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> >, char>' requested here
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/pinyin.cc:1:
In file included from src/pinyin.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:479:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:14:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iterator:421:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__functional_base:14:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:423:45: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &, const char &>' requested
      here
            _CheckArgsDep<_Dummy>::template __enable_implicit<_T1 const&, _T2 const&>()
                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:426:5: note: while substituting prior template arguments
      into non-type template parameter [with _Dummy = true]
    pair(_T1 const& __t1, _T2 const& __t2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _Dummy = (no value), $1 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/pinyin.cc:1:
In file included from src/pinyin.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:479:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:14:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iterator:421:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__functional_base:14:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, char const (&)[3]>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:441:34: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<char const (&)[3], char>' requested here
            _CheckArgs::template __enable_implicit<_U1, _U2>()
                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:444:5: note: while substituting prior template arguments
      into non-type template parameter [with _U1 = char const (&)[3], _U2 = char]
    pair(_U1&& __u1, _U2&& __u2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _U1 = char const (&)[3], _U2 = char, $2 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/pinyin.cc:1:
src/pinyin.h:16:53: error: no matching constructor for initialization of 'const std::map<std::string, char>' (aka 'const map<basic_string<char, char_traits<char>,
      allocator<char> >, char>')
  const std::map<std::string, char> tone_to_plain = {
                                                    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:980:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:989:9: note: candidate constructor template not viable:
      requires 4 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:999:5: note: candidate constructor template not viable:
      requires 3 arguments, but 29 were provided
    map(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1053:5: note: candidate constructor not viable: requires 3
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:975:14: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    explicit map(const key_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1035:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(map&& __m, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1046:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1061:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1081:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(const map& __m, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:968:14: note: candidate constructor not viable: requires
      single argument '__comp', but 29 arguments were provided
    explicit map(const key_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1004:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(const map& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1029:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(map&& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1075:14: note: candidate constructor not viable: requires
      single argument '__a', but 29 arguments were provided
    explicit map(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:960:5: note: candidate constructor not viable: requires 0
      arguments, but 29 were provided
    map()
    ^
In file included from src/pinyin.cc:1:
src/pinyin.h:23:47: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_prefix = {
                                              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 63 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 63 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 63 were provided
    set()
    ^
In file included from src/pinyin.cc:1:
src/pinyin.h:45:46: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_valid = {
                                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 399 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 399 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 399 were provided
    set()
    ^
6 errors generated.
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:657:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:305:9: error: implicit instantiation of undefined template
      'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    _T1 first;
        ^
src/pinyin.h:17:7: note: in instantiation of template class 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> >, char>' requested here
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:423:45: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &, const char &>' requested
      here
            _CheckArgsDep<_Dummy>::template __enable_implicit<_T1 const&, _T2 const&>()
                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:426:5: note: while substituting prior template arguments
      into non-type template parameter [with _Dummy = true]
    pair(_T1 const& __t1, _T2 const& __t2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _Dummy = (no value), $1 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:4:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:652:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:2803:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    : public integral_constant<bool, __is_constructible(_Tp, _Args...)>
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:360:18: note: in instantiation of template class
      'std::__1::is_constructible<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, char const (&)[3]>' requested here
          return is_constructible<first_type, _U1>::value
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:441:34: note: in instantiation of function template
      specialization 'std::__1::pair<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >,
      char>::_CheckArgs::__enable_implicit<char const (&)[3], char>' requested here
            _CheckArgs::template __enable_implicit<_U1, _U2>()
                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/utility:444:5: note: while substituting prior template arguments
      into non-type template parameter [with _U1 = char const (&)[3], _U2 = char]
    pair(_U1&& __u1, _U2&& __u2)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/pinyin.h:17:7: note: while substituting deduced template arguments into function template 'pair' [with _U1 = char const (&)[3], _U2 = char, $2 = (no value)]
      {"ā", 'a'}, {"á", 'a'}, {"ǎ", 'a'}, {"à", 'a'}, {"ē", 'e'}, {"é", 'e'}, {"ě", 'e'}, {"è", 'e'},
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:209:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:16:53: error: no matching constructor for initialization of 'const std::map<std::string, char>' (aka 'const map<basic_string<char, char_traits<char>,
      allocator<char> >, char>')
  const std::map<std::string, char> tone_to_plain = {
                                                    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:980:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:989:9: note: candidate constructor template not viable:
      requires 4 arguments, but 29 were provided
        map(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:999:5: note: candidate constructor template not viable:
      requires 3 arguments, but 29 were provided
    map(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1053:5: note: candidate constructor not viable: requires 3
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:975:14: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    explicit map(const key_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1035:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(map&& __m, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1046:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 29 were provided
    map(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1061:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1081:5: note: candidate constructor not viable: requires 2
      arguments, but 29 were provided
    map(const map& __m, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:968:14: note: candidate constructor not viable: requires
      single argument '__comp', but 29 arguments were provided
    explicit map(const key_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1004:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(const map& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1029:5: note: candidate constructor not viable: requires
      single argument '__m', but 29 arguments were provided
    map(map&& __m)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:1075:14: note: candidate constructor not viable: requires
      single argument '__a', but 29 arguments were provided
    explicit map(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:960:5: note: candidate constructor not viable: requires 0
      arguments, but 29 were provided
    map()
    ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:23:47: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_prefix = {
                                              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 63 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 63 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 63 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 63 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 63 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 63 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 63 were provided
    set()
    ^
In file included from src/simple_tokenizer.cc:1:
In file included from src/simple_tokenizer.h:5:
src/pinyin.h:45:46: error: no matching constructor for initialization of 'const std::set<std::string>' (aka 'const set<basic_string<char, char_traits<char>,
      allocator<char> > >')
  const std::set<std::string> pinyin_valid = {
                                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:505:9: note: candidate constructor template not viable:
      requires at most 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:514:9: note: candidate constructor template not viable:
      requires 4 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const value_compare& __comp,
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:524:9: note: candidate constructor template not viable:
      requires 3 arguments, but 399 were provided
        set(_InputIterator __f, _InputIterator __l, const allocator_type& __a)
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:571:5: note: candidate constructor not viable: requires 3
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:501:14: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    explicit set(const value_compare& __comp, const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:554:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(const set& __s, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:561:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(set&& __s, const allocator_type& __a);
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:564:5: note: candidate constructor not viable: requires at
      most 2 arguments, but 399 were provided
    set(initializer_list<value_type> __il, const value_compare& __comp = value_compare())
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:580:5: note: candidate constructor not viable: requires 2
      arguments, but 399 were provided
    set(initializer_list<value_type> __il, const allocator_type& __a)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:494:14: note: candidate constructor not viable: requires
      single argument '__comp', but 399 arguments were provided
    explicit set(const value_compare& __comp)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:529:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(const set& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:544:5: note: candidate constructor not viable: requires single
      argument '__s', but 399 arguments were provided
    set(set&& __s)
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:550:14: note: candidate constructor not viable: requires
      single argument '__a', but 399 arguments were provided
    explicit set(const allocator_type& __a)
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:486:5: note: candidate constructor not viable: requires 0
      arguments, but 399 were provided
    set()
    ^
6 errors generated.
make: *** [libsimple.dylib] Error 1

--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,6 @@
 a.out
 .ccls*
 sqlite*
-*.so
+libsimple.*
 build/
 *.gch

--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ SHELL_FLAGS = \
 UNAME_S := $(shell uname -s)
 
 ifeq ($(UNAME_S),Darwin)
-	LIBNAME := libsimple.dylib
+	LIBNAME := libsimple.dylib	# change lib name to dylib is macOS
 	SHAREDNAME := sqlite3.dylib
 else
 	LIBNAME := libsimple.so
@@ -31,6 +31,7 @@ endif
 clean-lib:
 	rm -f ${LIBNAME} ${SHAREDNAME}
 
+# compile libsimple
 ${LIBNAME}: ${SHAREDNAME} clean-lib
 	${CPP} -std=c++14 src/*.c src/*.cc -shared -rdynamic -o ${LIBNAME} -I build/ -fPIC ${LIB} ${SHELL_FLAGS}
 
@@ -48,8 +49,12 @@ build/sqlite3.c: $(SQLITE_BASENAME).zip
 $(SQLITE_BASENAME).zip:
 	wget -N -c "$(SQLITE_URL)"
 
+# compile sqlite3-shell for testing
 sqlite3-shell: build/sqlite3.c
 	${CC} -lm build/sqlite3.c build/shell.c ${LIB} -o $@ ${SHELL_FLAGS}
+
+.DEFAULT_GOAL := all
+all: sqlite3-shell ${LIBNAME}
 
 clean:
 	rm -f "$(SQLITE_BASENAME).zip"
@@ -70,6 +75,3 @@ main:
 format:
 	clang-format-7 -i *.cc src/*.h src/*.cc src/*.c
 	cpplint --extensions=h,cc --linelength=120 --filter=-build/include,-legal/copyright,-readability/streams,-whitespace/comma --root=src src/*
-
-.DEFAULT_GOAL := all
-all: sqlite3-shell ${LIBNAME}

--- a/Makefile
+++ b/Makefile
@@ -29,10 +29,10 @@ else
 endif
 
 clean-lib:
-	rm -f libsimple.so
+	rm -f ${LIBNAME} ${SHAREDNAME}
 
 ${LIBNAME}: ${SHAREDNAME} clean-lib
-	${CPP} src/*.h src/*.cc src/*.c -shared -rdynamic -o ${LIBNAME} -I build/ -fPIC ${LIB} ${SHELL_FLAGS}
+	${CPP} -std=c++14 src/*.c src/*.cc -shared -rdynamic -o ${LIBNAME} -I build/ -fPIC ${LIB} ${SHELL_FLAGS}
 
 ${SHAREDNAME}: build/sqlite3.c
 	${CC} $^ -c -o $@ -fPIC
@@ -71,3 +71,5 @@ format:
 	clang-format-7 -i *.cc src/*.h src/*.cc src/*.c
 	cpplint --extensions=h,cc --linelength=120 --filter=-build/include,-legal/copyright,-readability/streams,-whitespace/comma --root=src src/*
 
+.DEFAULT_GOAL := all
+all: sqlite3-shell ${LIBNAME}

--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@ select fts5(?1);
 ```
 输出空不报错就说明没问题。如果不支持可以在项目根目录 `make sqlite3-shell` 本地编译一个用于测试。
 
-2. `.load libsimple` 加载 so 文件
+2. `.load libsimple` 加载编译好的库文件
 
 然后就可以使用了，具体的例子可以参考 [test.sql](./test.sql)
 
@@ -20,10 +20,9 @@ select fts5(?1);
 
 ### 编译相关
 
-使用支持 c++11 以上的编译器编译
+使用支持 c++14 以上的编译器编译
 
-- `make libsimple.so` 打包扩展
-- `make sqlite3-shell` 编译测试用的 sqlite3
+- `make` 打包扩展，它会编译 libsimple 库文件和 sqlite3-shell 用于测试
 - `make test` 运行 test.sql
 
 ### 代码

--- a/src/pinyin.cc
+++ b/src/pinyin.cc
@@ -43,10 +43,10 @@ std::set<std::string> PinYin::to_plain(const std::string &input) {
   return s;
 }
 
-std::map<int, std::vector<std::string>> PinYin::build_pinyin_map() {
+std::map<int, std::vector<std::string> > PinYin::build_pinyin_map() {
   std::ifstream pinyin_file("./contrib/pinyin.txt");
   std::string line;
-  std::map<int, std::vector<std::string>> pinyin;
+  std::map<int, std::vector<std::string> > pinyin;
   std::regex re{R"(U\+(\w+):\s+(\S+)\s+.*)"};
   std::smatch match;
   while (std::getline(pinyin_file, line)) {

--- a/src/pinyin.h
+++ b/src/pinyin.h
@@ -3,13 +3,14 @@
 
 #include <map>
 #include <set>
+#include <string>
 #include <vector>
 
 namespace simple_tokenizer {
 
 class PinYin {
  private:
-  std::map<int, std::vector<std::string>> pinyin;
+  std::map<int, std::vector<std::string> > pinyin;
   const std::vector<std::string> empty_vector;
 
   // clang-format off
@@ -100,7 +101,7 @@ class PinYin {
     };
   // clang-format on
   std::set<std::string> to_plain(const std::string &input);
-  std::map<int, std::vector<std::string>> build_pinyin_map();
+  std::map<int, std::vector<std::string> > build_pinyin_map();
   std::vector<std::string> toUtf8Array(const std::string &input);
   int codepoint(const std::string &u);
   std::vector<std::string> _split_pinyin(const std::string &input, int begin, int end);

--- a/test.sql
+++ b/test.sql
@@ -1,7 +1,7 @@
 -- Examples to use simple tokenizer
 
 -- load so file
-.load libsimple.so
+.load libsimple
 
 -- set tokenize to simple
 CREATE VIRTUAL TABLE t1 USING fts5(x, tokenize = "simple");

